elements/lisk-cryptography/src/bls.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright © 2021 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+
+import {
+	blsSign,
+	blsVerify,
+	blsKeyValidate,
+	blsAggregate,
+	blsKeyGen,
+	blsFastAggregateVerify,
+	blsSkToPk,
+	BLS_SUPPORTED,
+} from './bls_lib';
+import { tagMessage } from './message_tag';
+import { readBit, writeBit } from './utils';
+
+export { BLS_SUPPORTED };
+export const generatePrivateKey = blsKeyGen;
+export const getPublicKeyFromPrivateKey = blsSkToPk;
+export const validateKey = blsKeyValidate;
+
+export const signBLS = (
+	tag: string,
+	networkIdentifier: Buffer,
+	data: Buffer,
+	privateKey: Buffer,
+): Buffer => blsSign(privateKey, tagMessage(tag, networkIdentifier, data));
+
+export const verifyBLS = (
+	tag: string,
+	networkIdentifier: Buffer,
+	data: Buffer,
+	signature: Buffer,
+	publicKey: Buffer,
+): boolean => blsVerify(publicKey, tagMessage(tag, networkIdentifier, data), signature);
+
+export const createAggSig = (
+	publicKeysList: Buffer[],
+	pubKeySignaturePairs: { publicKey: Buffer; signature: Buffer }[],
+): { aggregationBits: Buffer; signature: Buffer } => {
+	let aggregationBits = Buffer.alloc(Math.ceil(publicKeysList.length / 8));
+	const signatures: Buffer[] = [];
+
+	for (const pair of pubKeySignaturePairs) {
+		signatures.push(pair.signature);
+		const index = publicKeysList.findIndex(key => key.equals(pair.publicKey));
+		aggregationBits = writeBit(aggregationBits, index, true);
```

Comment:
if you alloc above, maybe mutating is better here instead of copy? (ie: it doesn't write to the given Buffer